components/centraldashboard/public/components/main-page.js

```diff
@@ -205,6 +205,20 @@ export class MainPage extends PolymerElement {
     _isInsideOfIframe() {
         return window.location !== window.parent.location;
     }
+
+    /**
+     * Builds and returns an href value preserving the existing query string and
+     * hash.
+     * @param {string} href
+     * @return {string}
+     */
+    _buildHref(href) {
+        const current = new URL(window.location.href);
+        const url = new URL(href, window.location.origin);
```

Comment:
Why do we need this, isn't that implicit?
```suggestion
        const url = new URL(href);
```

Reply:
According to MDN, *If url is a relative URL, base is required, and will be used as the base URL. If url is an absolute URL, a given base will be ignored.* If we were getting the href from the anchor element's href property it would already be absolute, but since we are passing in a relative string we need to provide the base. The suggested change failed when I tried it.

Reply:
Ok, perhaps I misunderstood what this was doing. Thanks for following through!